app/src/main/java/com/pvp/app/ui/common/Components.kt

```diff
@@ -208,4 +220,204 @@ fun TextField(
+                ) {
+                    androidx.compose.material3.Button(
+                        content = confirmButtonContent,
+                        onClick = {
+                            onConfirmClick()
+
+                            showDialog = false
+                        }
+                    )
+                }
+            },
+            dismissButton = {
+                Box(
+                    contentAlignment = Alignment.BottomEnd
+                ) {
+                    androidx.compose.material3.Button(
+                        content = dismissButtonContent,
+                        onClick = {
+                            onDismiss()
+
+                            showDialog = false
+                        },
+                    )
+                }
+            }
+        )
+    }
+}
+
+@Composable
+fun IconButtonWithDialog(
+    modifier: Modifier = Modifier,
+    icon: ImageVector = Icons.Outlined.Edit,
+    iconSize: Dp = 20.dp,
+    confirmButtonContent: @Composable RowScope.() -> Unit = { Text("Confirm") },
+    dismissButtonContent: @Composable RowScope.() -> Unit = { Text("Dismiss") },
+    dialogTitle: @Composable () -> Unit = { Text("Dialog Title") },
+    dialogContent: @Composable () -> Unit = { Text("Dialog Content") },
+    onConfirmClick: () -> Unit = {},
+    onDismiss: () -> Unit = {}
+) {
+    var showDialog by remember { mutableStateOf(false) }
+
+    Box(
+        modifier = modifier
+    ) {
+        IconButton(
+            onClick = {
+                showDialog = true
+            },
+            modifier = Modifier.size(iconSize)
+        ) {
+            Icon(
+                imageVector = icon,
+                contentDescription = null
+            )
+        }
+    }
+
+    if (showDialog) {
+        AlertDialog(
+            title = dialogTitle,
+            text = dialogContent,
+            onDismissRequest = {
+                onDismiss()
+
+                showDialog = false
+            },
+            confirmButton = {
+                Box(
+                    contentAlignment = Alignment.BottomStart
+                ) {
+                    androidx.compose.material3.Button(
+                        content = confirmButtonContent,
+                        onClick = {
+                            onConfirmClick()
+
+                            showDialog = false
+                        }
+                    )
+                }
+
+            },
+            dismissButton = {
+                Box(
+                    contentAlignment = Alignment.BottomEnd
+                ) {
+                    androidx.compose.material3.Button(
+                        content = dismissButtonContent,
+                        onClick = {
+                            onDismiss()
+
+                            showDialog = false
+                        }
+                    )
+                }
+            }
+        )
+    }
+}
+
+@Composable
+fun UserInfoItem(
```

Comment:
What is `User` within this component exactly? And what is the purpose of this component? Is it to only showcase information? Then `InfoItem` should be sufficient. But if this also supports editing the value, name must be changed to `InfoEditableItem` or if you are more comfortable with other naming system - `EditableInfoItem`.